udata/core/dataset/api.py

```diff
@@ -310,6 +310,8 @@ def post(self, dataset, rid):
         ResourceEditPermission(dataset).test()
         resource = self.get_resource_or_404(dataset, rid)
         infos = self.handle_upload(dataset)
+        if resource.fs_filename is not None:
```

Comment:
maybe we should put that as the very latest operation in the route, in case anything else fails we don't want to be in an inconsistent state (same for community resources).